tests/Unit/Framework/ActionTesting.hpp

```diff
@@ -568,5 +692,20 @@ struct get_array_index<ActionTesting::MockArrayChare> {
   template <typename Component>
   using f = typename Component::array_index;
 };
+template <>
+struct get_array_index<ActionTesting::MockGroupChare> {
```

Comment:
Thinking about this more: Yes, I could avoid 4 specializations, but I don't see why you would get a compile-time error when trying to access `array_index`.  Couldn't someone specify in a Component a misspelled `using chare_type=MockArrayyyChare` and a correct `using array_index=size_t`, and then accessing array_index would work but there may be a more confusing compiler message elsewhere?

Reply:
Yeah, that's a reasonable point -- this probably does avoid some misbehavior earlier than it would if were just a generic template. I'm fine with it being left as-is.